tests/test_url_build.py

```diff
@@ -32,12 +32,27 @@ def test_build_with_scheme_and_host():
     assert u == URL("http://127.0.0.1")
 
 
-def test_build_with_port():
-    with pytest.raises(ValueError):
-        URL.build(port=8000)
-
-    u = URL.build(scheme="http", host="127.0.0.1", port=8000)
-    assert str(u) == "http://127.0.0.1:8000"
+@pytest.mark.parametrize(
+    ("port", "exc", "match"),
+    [
+        pytest.param(
+            8000,
+            ValueError,
+            r"""(?x)
+            ^
+            Can't build URL with "port" but without "host"\.
+            $
+            """,
```

Comment:
So this didn't match for some reason. Need to check why.
```python-traceback
  _______________________ test_build_with_port[port-only] ________________________
  [gw2] linux -- Python 3.7.17 /tmp/tmp.WTVCAbf0yq/venv/bin/python
  
  port = 8000, exc = <class 'ValueError'>
  match = '(?x)\n            ^\n            Can\'t build URL with "port" but without "host"\\.\n            $\n            '
  
      @pytest.mark.parametrize(
          ("port", "exc", "match"),
          [
              pytest.param(
                  8000,
                  ValueError,
                  r"""(?x)
                  ^
                  Can't build URL with "port" but without "host"\.
                  $
                  """,
                  id="port-only",
              ),
              pytest.param(
                  "", TypeError, r"^The port is required to be int\.$", id="port-str"
              ),
          ],
      )
      def test_build_with_port(port, exc, match):
          with pytest.raises(exc, match=match):
  >           URL.build(port=port)
  
  exc        = <class 'ValueError'>
  match      = '(?x)\n            ^\n            Can\'t build URL with "port" but without "host"\\.\n            $\n            '
  port       = 8000
  
  /project/tests/test_url_build.py:55: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  
  cls = <class 'yarl.URL'>
  
      @classmethod
      def build(
          cls,
          *,
          scheme="",
          authority="",
          user=None,
          ***
          host="",
          port=None,
          path="",
          query=None,
          query_string="",
          fragment="",
          encoded=False,
      ):
          """Creates and returns a new URL"""
      
          if authority and (user or password or host or port):
              raise ValueError(
                  'Can\'t mix "authority" with "user", "password", "host" or "port".'
              )
          if not isinstance(port, (int, type(None))):
              raise TypeError("The port is required to be int.")
          if port and not host:
  >           raise ValueError('Can\'t build URL with "port" but without "host".')
  E           ValueError: Can't build URL with "port" but without "host".
  
  authority  = ''
  cls        = <class 'yarl.URL'>
  encoded    = False
  fragment   = ''
  host       = ''
  password   = None
  path       = ''
  port       = 8000
  query      = None
  query_string = ''
  scheme     = ''
  user       = None
  
  ../venv/lib/python3.7/site-packages/yarl/_url.py:239: ValueError
  
  During handling of the above exception, another exception occurred:
  
  port = 8000, exc = <class 'ValueError'>
  match = '(?x)\n            ^\n            Can\'t build URL with "port" but without "host"\\.\n            $\n            '
  
      @pytest.mark.parametrize(
          ("port", "exc", "match"),
          [
              pytest.param(
                  8000,
                  ValueError,
                  r"""(?x)
                  ^
                  Can't build URL with "port" but without "host"\.
                  $
                  """,
                  id="port-only",
              ),
              pytest.param(
                  "", TypeError, r"^The port is required to be int\.$", id="port-str"
              ),
          ],
      )
      def test_build_with_port(port, exc, match):
          with pytest.raises(exc, match=match):
  >           URL.build(port=port)
  E           AssertionError: Regex pattern did not match.
  E            Regex: '(?x)\n            ^\n            Can\'t build URL with "port" but without "host"\\.\n            $\n            '
  E            Input: 'Can\'t build URL with "port" but without "host".'
  
  exc        = <class 'ValueError'>
  match      = '(?x)\n            ^\n            Can\'t build URL with "port" but without "host"\\.\n            $\n            '
  port       = 8000
  
  /project/tests/test_url_build.py:55: AssertionError
```
```console
  FAILED ../../../project/tests/test_url_build.py::test_build_with_port[port-only] - AssertionError: Regex pattern did not match.
```

Reply:
> Whitespace within the pattern is ignored, except when in a character class, or when preceded by an unescaped backslash,

Reply:
Ah.. Thanks!